drivers/hmis/app/graphql/mutations/ac_hmis/update_referral_posting.rb

```diff
@@ -48,6 +49,21 @@ def resolve(id:, input:)
           end
           raise ActiveRecord::Rollback if errors.any?
 
+          # This is similar to the logic in query_type.rb:can_project_accept_referral, but differs in a couple ways:
+          # - Here, we do use viewable_by when querying Enrollments. (We already threw an error above if the user doesn't have this permission)
+          # - We return information about each conflicting enrollment, so the user can fix the errors.
+          personal_ids = Hmis::Hud::Client.where(id: posting.referral.household_members.pluck(:client_id)).pluck(:PersonalID)
+          # use project_pk so that wip enrollments are included
+          conflicting_enrollments = Hmis::Hud::Enrollment.viewable_by(current_user).where(project_pk: posting.project.id, personal_id: personal_ids)
```

Comment:
We can just use `project` to hide these implementation details. The `project` belongs_to relation uses the project_pk under the hood.
```
          conflicting_enrollments = Hmis::Hud::Enrollment.viewable_by(current_user).where(project: posting.project, personal_id: personal_ids)
```

Although, I think it would be clearer to just directly query the enrollments on the project. There is no need to check viewable_by because we already raise an error if the user cant view enrollments in the receiving project.

```suggestion
          conflicting_enrollments = posting.project.enrollments.open_including_wip.where(personal_id: personal_ids)
```
It also looks like this would error if the client had an exited enrollment. Adding `open_including_wip` above should address that.

Reply:
Maybe a further improvement, to avoid the plucking, would be:
```
household_members = posting.referral.household_members.map(&:client)
conflicting_enrollments = posting.project.enrollments.open_including_wip.where(client: household_members)
```

Reply:
I think the error here is with the enrollment <> client relationship, I'd like to understand better what is going on so I'll dig a bit more.
```
Error: the evaluation of `posting.project.enrollments.open_including_wip.where(client: household_members)` failed with the exception '[:data_source_id, :PersonalID] is not a symbol nor a string
```

Reply:
Ah okay, my bad. I thought our composite key relations supported that, but I guess they don't. I'm seeing the same behavior as you.

Reply:
Sorry for the confusion 😅

Reply:
Ah I see :( thanks for looking into it though!